### ISSUE TITLE
Python package has no dependencies (pip)

### ISSUE THREAD

**Issue by eldipa** [OfflineIMAP/offlineimap#661]
#### General informations

- system/distribution (with version): -
- offlineimap version (`offlineimap -V`): 7.3.3
- Python version: 2.7
- Pip version: pip 20.0.2

#### Logs, error

Installing offlineimap with pip (python 2) fails because the package try to use other
modules than are not added as dependencies.

```
ImportError: No module named six
ImportError: No module named rfc6555
```

#### Steps to reproduce the error

In empty Python 2 virtualenv, install offlineimap with pip:
```
pip install offlineimap
```

### Proposed solution

Add the missing packages to the `setup.py` of course but also decouple the `setup.py` of the `offlineimap` module.

I'm talking about this:
https://github.com/OfflineIMAP/offlineimap/blob/master/setup.py#L26
https://github.com/OfflineIMAP/offlineimap/blob/master/setup.py#L54-L59

Importing `offlineimap` to get those attributes forces `setup.py` to load everything including dependencies that may not be installed yet.

So in addition to adding the missing dependencies, another module could hold the metadata (https://github.com/OfflineIMAP/offlineimap/blob/master/offlineimap/__init__.py#L3-L13) separated from the rest of the code so it can be imported by `setup.py` without needing the dependencies to be installed.

Note: I'm not familiar with the building system of this project so I don't feel I could submit a patch. I promise that my next issue will come with the PR too.

**Comment by nicolas33** [comment]
You're correct about this cloupling issue. We currently support this installation process:
```
wget "https://raw.githubusercontent.com/OfflineIMAP/offlineimap/v7.3.3/requirements.txt" -O requirements.txt
pip install -r ./requirements.txt --user git+https://github.com/OfflineIMAP/offlineimap.git@v7.3.3
```

However, I agree we should apply the expected standards by correctly supporting just `pip install offlineimap`. 

About the resolution I'm not sure an out-of-tree module is a good practice. I wonder we should duplicate the data. Only one attribute is changing regularly (the __version__ attr). So, I'm fine to remove the import of offlineimap in setup.py and copy the attributes in this file.

I'd welcome your PR. ,-)

**Comment by eldipa** [comment]
The fix was applied to `next` branch (PR #664). Thanks to the reviewers for your time!

I'm closing this issue, reopen it if needed.
